Add notes about running Beats on Openshift (#3588)
elastic/cloud-on-k8s: docs/advanced-topics/openshift.asciidoc

elasticsearch-sample-es-dg4vvpm2mr scc:restricted
kibana-sample-kb-97c6b6b8d-lqfd2 scc:restricted
----

[id="{p}-openshift-beats"]
== Grant privileged permissions to Beats

> Deploying Beats on Openshift may require some privileged permissions. This section describes how to create a ServiceAccount, add the ServiceAccount to the `privileged` SCC and then use it to run Beats.

Reviewer: ```suggestion
Deploying Beats on Openshift may require some privileged permissions. This section describes how to create a ServiceAccount, add the ServiceAccount to the `privileged` SCC, and use it to run Beats.
```

> The following example assumes that Beats is deployed in a Namespace named `elastic` and we want to run Beats with a ServiceAccount named `filebeat`, replace these values according to your environment.

Reviewer: ```suggestion
The following example assumes that Beats is deployed in the Namespace `elastic` with the ServiceAccount `filebeat`. You can replace these values according to your environment.
```

> NOTE: If you have used the examples from the link:{eck_github}/tree/{eck_release_branch}/config/recipes/beats[recipes directory] the ServiceAccount may already exist.

Reviewer: ```suggestion
NOTE: If you used the examples from the link:{eck_github}/tree/{eck_release_branch}/config/recipes/beats[recipes directory], the ServiceAccount may already exist.
```

. Create a dedicated ServiceAccount
+
[source,shell]
----
oc create serviceaccount filebeat -n elastic
----
. Add the ServiceAccount to the required SCC
+
[source,shell]
----
oc adm policy add-scc-to-user privileged -z filebeat -n elastic
----
. Update the Beat manifest to use the new ServiceAccount, for example:
+
[source,yaml,subs="attributes"]
----
apiVersion: beat.k8s.elastic.co/v1beta1
kind: Beat
metadata:
  name: heartbeat
spec:
  type: heartbeat
  version: {version}
  elasticsearchRef:
    name: elasticsearch
  config:
    heartbeat.monitors:
    - type: tcp
      schedule: '@every 5s'
      hosts: ["elasticsearch-es-http.default.svc:9200"]
    - type: tcp
      schedule: '@every 5s'
      hosts: ["kibana-kb-http.default.svc:5601"]
  deployment:
    replicas: 1
    podTemplate:
      spec:
        serviceAccountName: filebeat
        securityContext:
          runAsUser: 0
----

If SELinux is enabled the Beat Pod may fail with the following message:

[source,shell]
----
Exiting: Failed to create Beat meta file: open /usr/share/heartbeat/data/meta.json.new: permission denied
----

> In order to fix this error the label `svirt_sandbox_file_t` must be applied to the directory `/var/lib/elastic/heartbeat/heartbeat-data/` on the Kubernetes node:

Reviewer: ```suggestion
To fix this error, apply the label `svirt_sandbox_file_t` to the directory `/var/lib/elastic/heartbeat/heartbeat-data/` on the Kubernetes node:
```

[source,shell]
----
chcon -Rt svirt_sandbox_file_t /var/lib/elastic/heartbeat/heartbeat-data/
----

> This should be done on all the hosts where the heartbeat Pod can be deployed.

Reviewer: ```suggestion
Repeat these steps on all the hosts where the heartbeat Pod can be deployed.
```
